src/pdf/DriverOverview/DriverOverviewPdfButton.tsx

```diff
@@ -0,0 +1,190 @@
+import { logErrorReturnLogId } from "@/logger/logger";
+import { formatDateToDate } from "@/common/format";
+import { Dayjs } from "dayjs";
+import { ParcelsTableRow } from "@/app/parcels/getParcelsTableData";
+import { PdfDataFetchResponse } from "../common";
+
+interface DriverOverviewData {
+    driverName: string;
+    date: Date;
+    tableData: DriverOverviewTableData[];
+    message: string;
+}
+
+type ParcelsForDeliveryResponse =
+    | {
+          data: Schema["parcels"][];
+          error: null;
+      }
+    | {
+          data: null;
+          error: { type: ParcelsForDeliveryErrorType; logId: string };
+      };
+
+type ParcelsForDeliveryErrorType = "parcelFetchFailed";
+
+const getParcelsForDelivery = async (parcelIds: string[]): Promise<ParcelsForDeliveryResponse> => {
+    const { data, error } = await supabase.from("parcels").select().in("primary_key", parcelIds);
+    if (error) {
+        const logId = await logErrorReturnLogId("Error with fetch: Parcels", error);
+        return { data: null, error: { type: "parcelFetchFailed", logId: logId } };
+    }
+    return { data: data, error: null };
+};
+
+type ClientByIdResponse =
+    | {
+          data: Schema["clients"];
+          error: null;
+      }
+    | {
+          data: null;
+          error: { type: ClientByIdErrorType; logId: string };
+      };
+
+type ClientByIdErrorType = "clientFetchFailed";
+
+const getClientById = async (clientId: string): Promise<ClientByIdResponse> => {
+    const { data, error } = await supabase
+        .from("clients")
+        .select()
+        .eq("primary_key", clientId)
+        .single();
+    if (error) {
+        const logId = await logErrorReturnLogId(
+            `Error with fetch: Client with id ${clientId}`,
+            error
+        );
+        return { data: null, error: { type: "clientFetchFailed", logId: logId } };
+    }
+    return { data: data, error: null };
+};
+
+const compare = (first: DriverOverviewTableData, second: DriverOverviewTableData): number => {
+    if (first.name < second.name) {
+        return -1;
+    }
+    if (first.name > second.name) {
+        return 1;
+    }
+    return 0;
+};
+
+type DriverPdfResponse =
+    | {
+          data: DriverOverviewTableData[];
+          error: null;
+      }
+    | {
+          data: null;
+          error: { type: DriverPdfErrorType; logId: string };
+      };
+
+type DriverPdfErrorType = ParcelsForDeliveryErrorType | ClientByIdErrorType;
+
+const getDriverPdfData = async (parcelIds: string[]): Promise<DriverPdfResponse> => {
+    const clientInformation = [];
+    const { data: parcels, error: parcelsError } = await getParcelsForDelivery(parcelIds);
+    if (parcelsError) {
+        return { data: null, error: parcelsError };
+    }
+    for (const parcel of parcels) {
+        const { data: client, error: clientError } = await getClientById(parcel.client_id);
+        if (clientError) {
+            return { data: null, error: clientError };
```

Comment:
Do you mean by using the parcels_plus view? I'm not sure it has all the information we want (address_1, address_2, delivery_instructions are missing) - I think we should change supabase call, and maybe others, to use views, but I thought this would be even further outside the scope of my ticket. (I didn't change the way these components fetch data other than the error handling).

Reply:
I thought you'd be able to it through the supabase call
https://supabase.com/docs/reference/javascript/select?queryGroups=example&example=query-referenced-tables